### DISCUSSION TITLE
How to add jars for Spark when submitting a spark job (JDBC)?

### DISCUSSION THREAD

**Discussion by HaoYang670** [apache/kyuubi#4486] [Q&A]
Currently, we are trying to add a customized Spark plugin jar file when submitting a Spark job.
We've tried:
1. Adding `spark.jars=***` into the JDBC connection URL, which doesn't work.
2. Executing the SQL `ADD JAR ***` during execution, which could add the jar file. But we also need to set some relative configs which can only be modified in spark-submit. So this way isn't suitable for us.

How could we add jars in spark-submit?

**Comment by zwangsheng** [comment]
Depends on how it is used, (and you should ensure that spark can get the files on filesystem, if deploy-mode is cluster)
1. kyuubi on kubernetes, we recommands upload jars to hdfs or other filesystem, and add `spark.jars=hdfs://`
2. kyuubi on physical machines, you can use `spark.jar=/path/to/`

**Reply by HaoYang670** [reply]
Yes we are using Spark on K8s (JDBC mode) and the jar is on s3 bucket. But  `spark.jars=...` doesn't work. We can't find the Class in that jar.

**Reply by zwangsheng** [reply]
You can look for this type of log `Added $jarMessage $path at ${added.mkString(",")} with timestamp $timestamp` in Spark Driver Log to see if it is added to the success
